src/dodal/devices/zocalo/zocalo_results.py

```diff
@@ -0,0 +1,224 @@
+                )
+
+            task = asyncio.create_task(_get_results())
+            try:
+                await asyncio.wait_for(asyncio.shield(task), self.timeout_s / 2)
+            except TimeoutError:
+                LOGGER.warning("Waited half of timeout for zocalo results - retrying")
+                await asyncio.wait_for(task, self.timeout_s / 2)
+
+        except TimeoutError:
+            LOGGER.warning("Timed out waiting for zocalo results!")
+            raise
+        finally:
+            self._kickoff_run = False
+
+    async def describe(self) -> dict[str, Descriptor]:
+        return OrderedDict(
+            [
+                (
+                    self._name + "-results",
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "array",
+                        "shape": [
+                            -1,
+                        ],  # TODO describe properly - see https://github.com/DiamondLightSource/dodal/issues/253
+                    },
+                ),
+                (
+                    self._name + "-centres_of_mass",
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "array",
+                        "shape": [-1, 3],
+                    },
+                ),
+                (
+                    self._name + "-bbox_sizes",
+                    {
+                        "source": f"sim://{self._prefix}",
+                        "dtype": "number",
+                        "shape": [-1, 3],
+                    },
+                ),
+            ],
+        )
+
+    def _get_zocalo_connection(self):
+        zc = zocalo.configuration.from_file()
+        zc.activate_environment(self.zocalo_environment)
+
+        transport = lookup("PikaTransport")()
+        transport.connect()
+        return transport
+
+    def _subscribe_to_results(self):
+        transport = self._get_zocalo_connection()
+
+        def _receive_result(
+            rw: workflows.recipe.RecipeWrapper, header: dict, message: dict
+        ) -> None:
+            LOGGER.info(f"Received {message}")
+            recipe_parameters = rw.recipe_step["parameters"]  # type: ignore # this rw is initialised with a message so recipe step is not None
+            LOGGER.info(f"Recipe step parameters: {recipe_parameters}")
+            transport.ack(header)
+
+            results = message.get("results", [])
+            call_in_bluesky_event_loop(self._raw_results_received.put(results))
+
+        subscription = workflows.recipe.wrap_subscribe(
+            transport,
+            self.channel,
+            _receive_result,
+            acknowledgement=True,
+            allow_non_recipe_messages=False,
+        )
+        LOGGER.info(f"Made zocalo queue subscription: {bool(subscription)}.")
+
+
+ZOCALO_READING_PLAN_NAME = "zocalo reading"
+
+
+def trigger_wait_and_read_zocalo(zocalo: ZocaloResults):
```

Comment:
Nit: Can we just use https://nsls-ii.github.io/bluesky/generated/bluesky.plan_stubs.trigger_and_read.html?